### ISSUE TITLE
Datasets compatibility?

### ISSUE THREAD

**Issue by hzh8311** [1adrianb/2D-and-3D-face-alignment#22]
When train the model on 300W-LP, the `center` and `scale` are predefined at [lines](https://github.com/1adrianb/face-alignment-training/blob/master/dataset-images.lua#L18-L19). But when I try to run you demo code I found that the `center` and `scale` are computed according to ground-truth landmarks [lines](https://github.com/1adrianb/2D-and-3D-face-alignment/blob/master/utils.lua#L184-L187). And I don't unstand how the parameters like 0.12 and 195 comes?

The AUC on LS3D-W balanced dataset is 17-18% lower than the result reported in the papers

**Comment by 1adrianb** [comment]
Hi @hzh8311, The two numbers are used to aproximatevly map the center and the scale obtained from a tight bounding box to the one used during training.

Regarding the results: You should get around 74% if you run this code. For LS3D-W balanced you have to remove the 1.75 shift at this line: https://github.com/1adrianb/2D-and-3D-face-alignment/blob/fc4d29999a1f0b8a833805bfe437bd32df886242/main.lua#L116 otherwise you will ge the wrong result. The shift is required only for 300W where there is a diference of indexes been 0 or 1 based.

**Comment by hzh8311** [comment]
So why 1.75 here? Should not it be 1. ?

**Comment by hzh8311** [comment]
It seems that the `center` and `scale` for training are fixed as (225, 275) and 1.8, respectively. And the `center` and `scale` for testing are computed according to bounding box. Does this difference matters and leads to such performance drop (17-18%)? @1adrianb

**Comment by 1adrianb** [comment]
@hzh8311 the center is fixed for 300W-LP because the images are already normalised and therefore computing a bounding box is not required. At test time however the faces are coming in all sort of sizes and are found at various locations in the image, therefore some sort of normalisation is required. This is done for simplicity based on the bounding box size.

The error of 17% is not coming from normalisation diff, but from the shift between the ground truth and the predictions. The shift is applied after the predictions are obtained.

**Comment by hzh8311** [comment]
I removed the shift you mentioned, but only 3-4% improvement compared to my first  result. That is to say it is still 13-14% lower than the result reported in the paper.
For more details, the AUCs on traning set and validation set(300W_LP_test) are quiet high (~80%), but it drop badly on LS3D-W dataset.

**Comment by 1adrianb** [comment]
Oh, so you are using a retrained model? Because using the pretained model, already provided, you should be able to match exactly that numbers.

**Comment by hzh8311** [comment]
Yes, but I can't figure out why the retrained model can not reach your performance.
